x-pack/apm-server/aggregation/lsm.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package aggregation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.uber.org/zap"
+
+	"github.com/elastic/apm-aggregation/aggregationpb"
+	"github.com/elastic/apm-aggregation/aggregators"
+	"github.com/elastic/apm-data/model/modelpb"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type Aggregator struct {
+	logger         *zap.Logger
+	baseaggregator *aggregators.Aggregator
+}
+
+// NewAggregator returns a new instance of aggregator.
+func NewAggregator(
+	ctx context.Context,
+	maxSvcs, maxTxGroups, maxSvcTxGroups, maxSpanGroups int,
+	nextProcessor modelpb.BatchProcessor,
+	logger *logp.Logger,
+) (*Aggregator, error) {
+	zapLogger := zap.New(logger.Core(), zap.WithCaller(true)).Named("aggregator")
+
+	baseaggregator, err := aggregators.New(
+		aggregators.WithLimits(aggregators.Limits{
+			MaxSpanGroups:                         maxSpanGroups,
+			MaxSpanGroupsPerService:               max(maxSpanGroups/10, 1),
+			MaxTransactionGroups:                  maxTxGroups,
+			MaxTransactionGroupsPerService:        max(maxTxGroups/10, 1),
+			MaxServiceTransactionGroups:           maxSvcTxGroups,
+			MaxServiceTransactionGroupsPerService: max(maxSvcTxGroups/10, 1),
+			MaxServiceInstanceGroupsPerService:    max(maxSvcs/10, 1),
+			MaxServices:                           maxSvcs,
+		}),
+		aggregators.WithProcessor(wrapNextProcessor(nextProcessor)),
+		aggregators.WithAggregationIntervals([]time.Duration{time.Minute, 10 * time.Minute, time.Hour}),
+		aggregators.WithLogger(zapLogger),
+		aggregators.WithMeter(otel.GetMeterProvider().Meter("aggregator")),
+		aggregators.WithTracer(otel.GetTracerProvider().Tracer("aggregator")),
+		aggregators.WithInMemory(true),
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create base aggregator: %w", err)
+	}
+	agg := &Aggregator{
+		logger:         zapLogger,
+		baseaggregator: baseaggregator,
+	}
+	return agg, err
+}
+
+// Run runs all the components of aggregator.
+func (a *Aggregator) Run() error {
+	return a.baseaggregator.Run(context.TODO())
```

Comment:
```suggestion
	return a.baseaggregator.Run(context.Background())
```

Also should we be checking that the result is not `ErrAggregatorClosed`? IIANM that's an expected result, if Close is called.